File: backend/src/main/java/com/corhuila/BackendSQL/Entity/Order.java
package com.corhuila.BackendSQL.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "customer_order")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate orderDate;

    @Column(nullable = false, length = 20)
    private String status;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    private Client client;

    @ManyToMany
    @JoinTable(
            name = "order_product",
            joinColumns = @JoinColumn(name = "order_id"),
            inverseJoinColumns = @JoinColumn(name = "product_id")
    )
    private List<Product> products;
}

File: frontend/src/app/model/order.ts
import { Product } from './product';
import { Customer } from './customer';

export class Order {
  id!: number;
  orderDate!: string;
  status!: string;
  totalAmount!: number;
  customer!: Customer;
  products!: Product[];

  constructor(orderDate: string, status: string, totalAmount: number, customer: Customer, products: Product[]) {
    this.orderDate = orderDate;
    this.status = status;
    this.totalAmount = totalAmount;
    this.customer = customer;
    this.products = products;
  }
}